src/spi.rs

```diff
@@ -1923,16 +2112,32 @@ extern "C" fn spi_notify(transaction: *mut spi_transaction_t) {
     }
 }
 
-fn copy_operation<'b>(operation: &'b mut Operation<'_, u8>) -> Operation<'b, u8> {
+fn copy_operation<'b>(operation: &'b mut Operation<'_>) -> Operation<'b> {
```

Comment:
Nit: I don't think you need `operation: &'b mut <Operation<'_>` to be mutable.

Reply:
Nit 2: Wouldn't it be cleaner if you implement `Clone` on our operation?

Reply:
Heh. I just realized I'm reviewing my own code :-)

Reply:
For both of these they need to be `&mut` so that we can put the `&mut buffer` into the read variants. And also it can't be `Clone` because we can't consume the input Operation, as we're passed it via `&mut`.